parser/helper.go

```diff
@@ -79,10 +80,15 @@ func (p *parserHelper) newLiteralDouble(ctx interface{}, value float64) *exprpb.
 		&exprpb.Constant{ConstantKind: &exprpb.Constant_DoubleValue{DoubleValue: value}})
 }
 
-func (p *parserHelper) newIdent(ctx interface{}, name string) *exprpb.Expr {
+func (p *parserHelper) newIdent(ctx interface{}, name string) (*exprpb.Expr, error) {
+	for _, r := range reservedIds {
+		if r == name {
+			return nil, fmt.Errorf("reserved identifier: %s", name)
```

Comment:
Since all calls eventually resolve to `newIdent()` within the parser helper, have you considered reporting the error here rather than returning it? This would mean that you would need to wait until after the `newExpr` call in order to ensure you have an expression id that can be mapped to a source position.

Reply:
Mooted by move to visitor.